lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -3215,9 +3215,12 @@ bool SROA_Helper::DoScalarReplacement(GlobalVariable *GV,
 }
 
 static void ReplaceConstantWithInst(Constant *C, Value *V, IRBuilder<> &Builder) {
+  Function *F = Builder.GetInsertBlock()->getParent();
```

Comment:
Should we be grabbing the function from the value V (in the case where V is an instruction). There is no obvious guarantee from this context that V was created by builder at that location, but maybe it is enforced by the caller? 

Seems to be a potentially fragile assumption.

Reply:
The builder assumption must be true, it's what's used to create instruction replacements for constant expressions, like GEP.  It's based on the location of the memcpy instruction being replaced.

Reply:
For extra clarity: We are attempting to replace a memcpy instruction.  Any replacements should only occur within the same function as that memcpy instruction.  The builder is based on that memcpy instruction location.  If V is constant, I still don't think we should be replacing users in functions other than the one with the memcpy instruction being replaced.  If we conditionally skipped only when V was an instruction and in a different function, we would replace users of constants in other functions.  I'm just not positive that constant user replacement in other functions is what we want here.